core/src/main/java/com/google/zxing/common/StringUtils.java

```diff
@@ -29,28 +30,27 @@
  */
 public final class StringUtils {
 
-  private static final String PLATFORM_DEFAULT_ENCODING = Charset.defaultCharset().name();
-  public static final String SHIFT_JIS = "SJIS";
-  public static final String GB2312 = "GB2312";
-  private static final String EUC_JP = "EUC_JP";
-  private static final String UTF8 = "UTF8";
-  private static final String ISO88591 = "ISO8859_1";
+  private static final Charset PLATFORM_DEFAULT_ENCODING = Charset.defaultCharset();
+  public static final Charset SHIFT_JIS = Charset.forName("SJIS");
```

Comment:
This gets flagged by the binary compatibility check. While it's probably in practice not a field anyone uses, let's keep the existing public symbols for full compatibility and introduce new ones, even if the old ones are then superfluous.

Reply:
(These were Strings originally simply because this originally had to work on J2ME, Java 1.3 or something, well before Charset. I agree Charset is better)

Reply:
Aha, makes sense. Will fix.